TPP.Core/Commands/Definitions/BadgeCommands.cs

```diff
@@ -339,5 +358,44 @@ public async Task<CommandResult> GiftBadge(CommandContext context)
                 ResponseTarget = ResponseTarget.Chat
             };
         }
+
+        public async Task<CommandResult> ListSellBadge(CommandContext context)
+        {
+            (Optional<User> userOpt, PkmnSpecies species, Optional<Badge.BadgeSource> sourceOpt, Optional<Shiny> shinyOpt, Optional<Form> formOpt) =
+                await context.ParseArgs<AnyOrder<Optional<User>, PkmnSpecies, Optional<Badge.BadgeSource>, Optional<Shiny>, Optional<Form>>>();
+
+            User user = userOpt.IsPresent ? userOpt.Value : context.Message.User;
+            (Badge.BadgeSource? source, string? form, bool? shiny) = InterpretBadgeInfoArgs(sourceOpt, formOpt, shinyOpt);
+
+            List<Badge> forSale = await _badgeRepo.FindAllForSaleByCustom(user.Id, species, form, source, shiny);
+
+            string response;
+            if (forSale.Count == 0)
+                response = "No badges found.";
+            else
+            {
+                response = string.Format("{0} badges found:", forSale.Count);
+                foreach (Badge b in forSale)
+                {
+                    if (b.UserId == null)
+                        throw new OwnedBadgeNotFoundException(b);
+                    User? seller = await _userRepo.FindById(b.UserId);
+                    if (seller == null)
+                        throw new OwnedBadgeNotFoundException(b);
+
+                    response += string.Format(" {0}{1}{2} sold by {3} for T{4}",
+                        shiny == true ? "Shiny " : "",
+                        form ?? "",
+                        b.Species.Name,
+                        seller.SimpleName,
+                        b.SellPrice);
```

Comment:
avoid building strings in a loop with `+=`. Instead, Build an enumerable and `string.Join` it.